File: src/main/java/ru/mpu/horse_races/services/JockeyService.java
package ru.mpu.horse_races.services;

import ru.mpu.horse_races.domain.dtos.CreateOrUpdateJockeyDtoRq;
import ru.mpu.horse_races.domain.dtos.JockeyDto;

import java.util.List;

public interface JockeyService {
    JockeyDto insert(CreateOrUpdateJockeyDtoRq jockey);

    JockeyDto update(CreateOrUpdateJockeyDtoRq jockey);

    List<JockeyDto> findAll();

    JockeyDto findById(Long id);
}

File: frontend/horse-races/src/app/jockey/jockey.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import {CreateOrUpdateJockeyDtoRq, JockeyDto} from "./jockey.models";

@Injectable({
    providedIn: 'root'
})
export class JockeyService {
    private apiUrl = '/api/v1/horse'

    constructor(private http: HttpClient) {}

    findAll(): Observable<JockeyDto[]> {
        return this.http.get<JockeyDto[]>(this.apiUrl);
    }

    insert(jockey: CreateOrUpdateJockeyDtoRq): Observable<CreateOrUpdateJockeyDtoRq> {
        return this.http.post<CreateOrUpdateJockeyDtoRq>(this.apiUrl, jockey);
    }

    update(jockey: CreateOrUpdateJockeyDtoRq): Observable<CreateOrUpdateJockeyDtoRq> {
        return this.http.put<CreateOrUpdateJockeyDtoRq>(this.apiUrl, jockey);
    }

    deleteById(id: number): Observable<void> {
        return this.http.delete<void>(`${this.apiUrl}/${id}`);
    }
}
